apiserver/facades/client/bundle/bundle_test.go

```diff
@@ -1066,6 +1066,55 @@ applications:
 	s.st.CheckCall(c, 0, "ExportPartial", s.st.GetExportConfig())
 }
 
+func (s *bundleSuite) TestExportBundleWithApplicationStorage(c *gc.C) {
+	s.st.model = description.NewModel(description.ModelArgs{Owner: names.NewUserTag("magic"),
+		Config: map[string]interface{}{
+			"name": "awesome",
+			"uuid": "some-uuid",
+		},
+		CloudRegion: "some-region"})
+
+	args := s.minimalApplicationArgs(description.IAAS)
+	// Add storage constraints to the app
+	args.StorageConstraints = map[string]description.StorageConstraintArgs{
+		"storage1": {
+			Pool:  "pool1",
+			Size:  1024,
+			Count: 3,
+		},
+	}
+	app := s.st.model.AddApplication(args)
+	app.SetStatus(minimalStatusArgs())
+
+	u := app.AddUnit(minimalUnitArgs(app.Type()))
+	u.SetAgentStatus(minimalStatusArgs())
+
+	s.st.model.SetStatus(description.StatusArgs{Value: "available"})
+
+	result, err := s.facade.ExportBundle(params.ExportBundleParams{})
+	c.Assert(err, jc.ErrorIsNil)
+	expectedResult := params.StringResult{Result: `
+series: trusty
+applications:
+  ubuntu:
+    charm: cs:trusty/ubuntu
+    channel: stable
+    num_units: 1
+    to:
+    - "0"
+    options:
+      key: value
+    storage:
+      storage1: pool1,3,1024
```

Comment:
Not sure, but would it be better / more future-proof to just serialize this as a YAML object? Like:

```
storage:
  storage1:
    pool: pool1
    size: 3
    count: 1024
```

Reply:
Yeah I do like that way more, I don't know if this is currently accepted by `juju deploy` tho?

Reply:
Unfortunately it seems that `juju deploy` doesn't accept this format in bundle yaml files. So I'm gonna leave it as is for now.